tests/test_namespace_matching.py

```diff
@@ -0,0 +1,155 @@
+        self.assertTrue(startswith_ns(("foo", "bar"), ("foo", "bar")))
+        self.assertFalse(startswith_ns(("foo",), ("foo", "bar")))
+        self.assertFalse(startswith_ns(("foobar",), ("foo",)))
+
+    def test_frozen_keymap_submap(self):
+        sep = NAMESPACE_SEP
+        data = {
+            "foo": 1,
+            f"foo{sep}bar": 2,
+            f"foo{sep}bar{sep}baz": 3,
+            "foobar": 4,
+            "other": 5
+        }
+        m = MyMap(data)
+        
+        # Test submap
+        sub = m.submap("foo")
+        self.assertIn("foo", sub)
+        self.assertIn(f"foo{sep}bar", sub)
+        self.assertIn(f"foo{sep}bar{sep}baz", sub)
+        self.assertNotIn("foobar", sub)
+        self.assertNotIn("other", sub)
+        
+        # Test exact match
+        sub2 = m.submap(f"foo{sep}bar")
+        self.assertIn(f"foo{sep}bar", sub2)
+        self.assertIn(f"foo{sep}bar{sep}baz", sub2)
+        self.assertNotIn("foo", sub2)
+
+    def test_flat_keys_values(self):
+        sep = NAMESPACE_SEP
+        data = {
+            "foo": 1,
+            f"foo{sep}bar": 2,
+            "other": 3
+        }
+        flat = Flat(data)
+        
+        # Test keys
+        keys = list(flat.keys("foo"))
+        self.assertIn(f"foo{sep}bar", keys)
+        self.assertIn("foo", keys)
+        self.assertNotIn("other", keys)
+        
+        # Test values
+        values = list(flat.values("foo"))
+        self.assertIn(1, values)
+        self.assertIn(2, values)
+        self.assertNotIn(3, values)
+
+    def test_namespace_walker_views(self):
+        sep = NAMESPACE_SEP
+        # Nested expects nested dicts.
+        # Nested keys are NOT namespaced strings in the input dict, but structure.
+        # But Nested.keys(namespace) returns namespaced keys.
+        n = Nested({"foo": {"bar": 1}})
+        
+        # Test keys view with namespace prefix
+        keys_foo = n.keys("foo")
+        # "foo" -> {"bar": 1}
+        # keys should be "foo:bar" (or "foo;bar")
+        self.assertIn(f"foo{sep}bar", keys_foo)
+        
+        # Test keys view with exact namespace match (should be empty for this data)
+        keys_foobar = n.keys(f"foo{sep}bar")
+        self.assertNotIn(f"foo{sep}bar", keys_foobar)
+        
+        # Test negative
+        keys_bar = n.keys("bar")
+        self.assertNotIn(f"foo{sep}bar", keys_bar)
+
+    def test_keymap_flat_keys_values(self):
+        sep = NAMESPACE_SEP
+        data = {
+            "foo": 1,
+            f"foo{sep}bar": 2,
+            "other": 3
+        }
+        flat = KeymapFlat(data)
+        
+        # Test keys
+        keys = list(flat.keys("foo"))
+        self.assertIn(f"foo{sep}bar", keys)
+        self.assertIn("foo", keys)
+        self.assertNotIn("other", keys)
+        
+        # Test values
+        values = list(flat.values("foo"))
+        self.assertIn(1, values)
+        self.assertIn(2, values)
+        self.assertNotIn(3, values)
+
```

Comment:
I super appreciate these tests, and they seem well-written – please consider updating these to our house-standard `py.test` syntax and/or using CLU's inline test harness!